client/state/themes/actions/activate-style-variation.ts

```diff
@@ -0,0 +1,27 @@
+import { getGlobalStylesId, updateGlobalStyles } from 'calypso/state/global-styles/actions';
+import { getCanonicalTheme } from 'calypso/state/themes/selectors';
+import type { GlobalStyles } from '@automattic/data-stores';
+import type { CalypsoDispatch } from 'calypso/state/types';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Triggers a network request to activate a specific style variation on a given site.
+ * @param {string}  themeId      Theme ID
+ * @param {number}  siteId       Site ID
+ * @param {Object}  globalStyles The global styles to be activated
+ * @returns {Function}           Action thunk
+ */
+export function activateStyleVariation(
+	themeId: string,
+	siteId: number,
+	globalStyles: GlobalStyles
+) {
+	return async ( dispatch: CalypsoDispatch, getState: AppState ) => {
+		const state = getState();
+		const theme = getCanonicalTheme( state, siteId, themeId );
+		const themeStylesheet = theme?.stylesheet || themeId;
```

Comment:
I found that it's weird to have this `stylesheet` <=> `themeId` in both here and in `getGlobalStylesId()`, can we just pass `themeId` to `getGlobalStylesId()` and then do the if-Jetpack logic there? 😄 So this function really is just a wrapper around `getGlobalStylesId()` and `updateGlobalStyles()`.

Reply:
Addressed via https://github.com/Automattic/wp-calypso/pull/86158/commits/90278056f9cc2db3a01ed3edec6b79e7368e111c